manifests/resource/vhost.pp

```diff
@@ -536,7 +537,9 @@
 
   # Check to see if SSL Certificates are properly defined.
   if ($ssl == true) {
-    if ($ssl_cert == undef) or ($ssl_key == undef) {
+    $cert_elsewhere = member($defined_elsewhere, 'ssl_cert')
+    $key_elsewhere =  member($defined_elsewhere, 'ssl_key')
+    if ($ssl_cert == undef and !$cert_elsewhere) or ($ssl_key == undef and !$key_elsewhere) {
       fail('nginx: SSL certificate/key (ssl_cert/ssl_key) and/or SSL Private must be defined and exist on the target system(s)')
```

Comment:
I think this `fail` should be removed, or downgraded to a warning with appropriate text (which we'll have to think about)